### DISCUSSION TITLE
socketIo server on SAMD board (nano iot)?

### DISCUSSION THREAD

**Discussion by sdetweil** [khoih-prog/WebSockets_Generic#23] [Q&A]
I have an environment where multiple Arduino devices send status and a mobile app also collects status. 3-4 times a second.

socketIo would make this a lot easier.  but, I need one board to be the server that every other thing connects to. it is also the wifi ap. 

I see clients, but no servers for SAMD. all the messages are JSON text, so ws would be ok too, altho I like the path triggering in socketIo.

**Comment by khoih-prog** [comment]
You can try WS Server for SAMD by using [**WebSockets2_Generic** Library](https://github.com/khoih-prog/WebSockets2_Generic). You can also use much better ESP32 as WS Server using this [**WebSockets_Generic**](https://github.com/khoih-prog/WebSockets_Generic) Library.

I don't know when the SIO Server will be implemented, for ESP32/ESP8266, etc. But I think you can use cheap **Raspberry Pi Zero W**, running Raspbian Linux and SIO server easily and reliably.

```
Linux raspberrypi-02 5.10.63-v7+ #1496 SMP Wed Dec 1 15:58:11 GMT 2021 armv7l GNU/Linux
```

**Reply by sdetweil** [reply]
yes, pi02w..  was just thinking the same. my reason for Arduino is gone.
